CruCentralCoast/Events/EventDetailsVC.swift

```diff
@@ -49,44 +57,113 @@ class EventDetailsVC: UIViewController {
     }
     
+    
+    @IBAction func locationButtonPressed(_ sender: Any) {
+        guard let eventLocation = self.eventLocation else {
+            print("error1")
+            return
+        }
+        
+        let geoCoder = CLGeocoder()
+        geoCoder.geocodeAddressString(eventLocation) { (placemarks, error) in
+            guard let clPlacemark = placemarks?.first
+            else {
+                print("location not found")
+                //handle no location found
+                return
+            }
+            
+            guard
+                let lat = clPlacemark.location?.coordinate.latitude,
+                let lon = clPlacemark.location?.coordinate.longitude
+            else {
+                return
+            }
+            let regionDistance: CLLocationDistance = 1000
+            let coordinates = CLLocationCoordinate2DMake(lat, lon)
+            let regionSpan = MKCoordinateRegionMakeWithDistance(coordinates, regionDistance, regionDistance)
+            let options = [
+                MKLaunchOptionsMapCenterKey: NSValue(mkCoordinate: regionSpan.center),
+                MKLaunchOptionsMapSpanKey: NSValue(mkCoordinateSpan: regionSpan.span)
+            ]
+            let mkPlacemark = MKPlacemark(coordinate: coordinates, addressDictionary: nil)
+            let mapItem = MKMapItem(placemark: mkPlacemark)
+            mapItem.name = self.eventViewableLocation
+            mapItem.openInMaps(launchOptions: options)
+        }
+    }
+    
+//    func getCoordinate(eventLocation: String) -> (CLLocationDegrees, CLLocationDegrees){
+//
+//        return (lat, lon)
+//    }
+    
+    //found at https://www.hackingwithswift.com/read/32/3/how-to-use-sfsafariviewcontroller-to-browse-a-web-page
     @IBAction func facebookButtonPressed(_ sender: Any) {
-        self.presentAlert(title: "Link not set up yet!!", message: "Coming soon")
+        guard let facebookURL = event?.facebookURL else {
+            return
+        }
+        
+        if let url = URL(string: facebookURL) {
+            let config = SFSafariViewController.Configuration()
+            config.entersReaderIfAvailable = true
+            
+            let vc = SFSafariViewController(url: url, configuration: config)
+            present(vc, animated: true)
+        }
     }
 
-    /* future function for calendar button on events Details Screen
-    @IBAction func calendarButtonPressed(_ sender: Any) {
+    @IBAction func addToCalendarButtonPressed(_ sender: Any) {
         let eventStore : EKEventStore = EKEventStore()
         
-        eventStore.requestAccess(to: .event, completion: { (granted, error) in
+        eventStore.requestAccess(to: .event) { (granted, error) in
             if (granted) && (error == nil) {
-                let event = EKEvent(eventStore: eventStore)
-                event.title = self.titleLabel.text
+                let event:EKEvent = EKEvent(eventStore: eventStore)
```

Comment:
I'd say be consistent throughout the project with how you place the spaces for these variable declarations. above on 136 you have `eventStore : EKEventStore` and here `event:EKEvent` my team's preferred way is the following:
`let event: EKEvent`